### PR TITLE
Dockerize. Special libertym build.

### PR DESCRIPTION
**Pull request by richard-sl**
@saucy-aseem @halkeye - Please review.

I am not sure we want to merge this whole thing in its entirety, but I think we should keep all this libertym stuff together.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.saucelabs.bamboo</groupId>
     <artifactId>bamboo-sauceondemand-plugin</artifactId>
-    <version>1.6.78-SNAPSHOT</version>
```
Comment (halkeye) [inline pom.xml]: Why the change of version?

Snapshot is pretty java standard. Does the maven release tools handle other phrases?

File: README.md [hunk 2]
```diff
@@ -2,9 +2,23 @@
...
+```
+docker-compose exec atlas-plugin-sdk bash
+```
+
+From that shell it should be possible to run the following commands:
+
+To build (compile,test,jar) the plugin:
 
 ```
 atlas-package
```
Comment (halkeye) [inline README.md]: These should be made to docker compose too

**Comment by halkeye** [comment]
All the .java files can probably be ignored.

I'm not a fan of docker for this sort of thing, but I guess it's totally optional anyways

File: pom.xml [hunk 3]
```diff
@@ -224,7 +224,6 @@
                 <configuration>
                     <productVersion>${bamboo.version}</productVersion>
                     <productDataVersion>${bamboo.data.version}</productDataVersion>
-                    <productDataPath>../generated-test-resources.zip</productDataPath>
```
Comment (halkeye) [inline pom.xml]: Removing this means you need to completely setup your test instance every time.

There are both good and bad points for this. It just gets very tiring to do every time you do a clean

**Comment by halkeye** [comment]
Also I think .78 might have already been released. Should be .79 snapshot

**Comment by richard-sl** [comment]
> All the .java files can probably be ignored.

I'll try omitting:
```
# Copy Maven preference files to predefine the command line question about # subscribing to the mailing list to `NO`. 
COPY .java /root/.java
```
From the Dockerfile to check that nothing blows up.

> I'm not a fan of docker for this sort of thing, but I guess it's totally optional anyways

I'm curious to know where you're coming from there. I think it'll save folk a bunch of time from setting up a local development environment manually, and helps guard against inadvertently installing different versions of the environmental dependencies.

File: pom.xml (hunk 3 above)
Comment (richard-sl) [inline pom.xml]: I recall being unable to get a build without commenting out this line. Might that have to do with using `atlas-run`'s build side effect instead of using `atlas-package`?

Comment (halkeye) [inline pom.xml]: I never committed the file cause it has my saucelabs credentials in it. I sent it to aseem by email.

you need to remove that line if you don't have the file

**Comment by halkeye** [comment]
> I'm curious to know where you're coming from there. I think it'll save folk a bunch of time from setting up a local development environment manually, and helps guard against inadvertently installing different versions of the environmental dependencies.

Mostly because it'll complicate IDEs because the deps will no longer be installed, so it can't really check import statements or auto complete or anything else.

File: pom.xml (hunk 1 above)
Comment (richard-sl) [inline pom.xml]: My intent was to call out the one-off-ness of this build. If we do a `1.6.79-SNAPSHOT` build I would think we want some way of denoting that it's different from the regular versions before/after it? Where would the better place to do that be?

**Comment by richard-sl** [comment]
> Mostly because it'll complicate IDEs because the deps will no longer be installed, so it can't really check import statements or auto complete or anything else.

Good point. That's hella annoying. Do you think some reverse mounts from within the container back out to the working directory would be good enough for the IDE's then?

File: README.md (hunk 2 above)
Comment (richard-sl) [inline README.md]: I'm not sure I follow? I thought you liked the idea of keeping the container stuff optional?

**Comment by richard-sl** [comment]
> All the .java files can probably be ignored.
> I'll try omitting ... [.java]

selection_008

Without the `.java` XML's the `atlas-*` commands can result in an annoying prompt that'll block further automation.
I don't think they can be ignored? Or were you thinking of something else?

**Comment by maciejurbanskisaucelabs** [comment]
Not merger for so long, so it's not needed ;p
